### ISSUE TITLE
Components not initially recognized in a .razor file in a Blazor Web App

### ISSUE THREAD

**Issue by danroth27** [microsoft/vscode-dotnettools#853]
Type: <b>Bug</b>

- Create a new Blazor Web App
- In the file explorer quickly open up Counter.razor
- Watch the `PageTitle` component to see if it is recognized as a component

Expected result: `PageTitle` turns green to indicate it is a Blazor component.
Actual result: `PageTitle` never turns green - it stays blue, event after waiting 30sec.
Workaround: Close and reopen the file. The `PageTitle` component is recognized almost immediately.

Extension version: 1.3.2
VS Code version: Code 1.85.1 (0ee08df0cf4527e40edc9aa28f4b5bd38bbff2b2, 2023-12-13T09:49:37.021Z)
OS version: Windows_NT x64 10.0.22621
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz (8 x 3600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|63.84GB (32.49GB free)|
|Process Argv|--crash-reporter-id 4f9325b6-0f0b-4512-95c8-0e78b95da0c9|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:30899288
vsclangdf:30486550
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30933248
pythongtdpath:30769146
i26e3531:30792625
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pyreplss1:30897532
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
accentitlementsc:30887149
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
aa_t_chat:30882232

```

</details>

**Comment by danroth27** [comment]
![Code_ozUXDpXsSG](https://github.com/microsoft/vscode-dotnettools/assets/1874516/c5069709-443a-4d09-9cee-183211a32ae3)

It looks like the tag is recognized as a complete (hover works) but it doesn't get colored correctly until you reopen the file.

**Comment by ryzngard** [comment]
This should be fixed in `v2.39.29`
